server/command.go

```diff
@@ -240,6 +249,168 @@ func (p *Plugin) executeShowCommand(args *model.CommandArgs) (*model.CommandResp
+						gotErrors = true
+					}
+
+					if mmTeam != nil {
+						mattermostTeamIDsVsNames[link.MattermostTeam] = mmTeam.DisplayName
+					}
+				}
+
+				// Get the Mattermost channel details
+				mmChannel, channelErr := p.API.GetChannel(link.MattermostChannel)
+				if channelErr != nil {
+					p.API.LogDebug("Unable to get the Mattermost channel information", "ChannelID", link.MattermostChannel, "Error", channelErr.DetailedError)
+					gotErrors = true
+					continue
+				}
+
+				mattermostChannelIDsVsNames[link.MattermostChannel] = mmChannel.DisplayName
+			}
+		}(idx)
+	}
+
+	wg.Wait()
+
+	// Get MS Teams display names for each unique team ID and store it
+	teamDetailsErr := p.GetMSTeamsTeamDetails(msTeamsTeamIDsVsNames)
+	gotErrors = gotErrors || teamDetailsErr
+
+	// Get MS Teams channel details for all channels for each unique team
+	channelDetailsErr := p.GetMSTeamsChannelDetailsForAllTeams(msTeamsTeamIDsVsChannelsQuery, msTeamsChannelIDsVsNames, &wg)
+	gotErrors = gotErrors || channelDetailsErr
+
+	for _, link := range links {
+		row := fmt.Sprintf(
+			"\n|%s|%s|%s|%s|",
+			mattermostTeamIDsVsNames[link.MattermostTeam],
+			mattermostChannelIDsVsNames[link.MattermostChannel],
+			msTeamsTeamIDsVsNames[link.MSTeamsTeam],
+			msTeamsChannelIDsVsNames[link.MSTeamsChannel],
+		)
+
+		if row != "\n|||||" {
+			sb.WriteString(row)
+		}
+	}
+
+	if gotErrors {
+		sb.WriteString("\nThere were some errors while fetching information about some links. Please check the server logs.")
+	}
+
+	p.sendBotEphemeralPost(userID, channelID, sb.String())
+}
+
+func (p *Plugin) GetMSTeamsTeamDetails(msTeamsTeamIDsVsNames map[string]string) bool {
+	var msTeamsFilterQuery strings.Builder
+	msTeamsFilterQuery.WriteString("id in (")
+	length := len(msTeamsTeamIDsVsNames)
+	count := 0
+	for teamID := range msTeamsTeamIDsVsNames {
+		count++
+		msTeamsFilterQuery.WriteString("'" + teamID + "'")
+		if count != length {
+			msTeamsFilterQuery.WriteString(", ")
+		}
+	}
+
+	msTeamsFilterQuery.WriteString(")")
+	msTeamsTeams, err := p.msteamsAppClient.GetTeams(msTeamsFilterQuery.String())
+	if err != nil {
+		p.API.LogDebug("Unable to get the MS Teams teams information", "Error", err.Error())
+		return true
+	}
+
+	for _, msTeamsTeam := range msTeamsTeams {
+		msTeamsTeamIDsVsNames[msTeamsTeam.ID] = msTeamsTeam.DisplayName
+	}
+
+	return false
+}
+
+func (p *Plugin) GetMSTeamsChannelDetailsForAllTeams(msTeamsTeamIDsVsChannelsQuery, msTeamsChannelIDsVsNames map[string]string, wg *sync.WaitGroup) bool {
+	gotErrors := false
+	for teamID, channelsQuery := range msTeamsTeamIDsVsChannelsQuery {
+		if wg != nil {
+			wg.Add(1)
+		}
+
+		go func(teamID, channelsQuery string) {
+			if wg != nil {
+				defer wg.Done()
+			}
+
+			channels, err := p.msteamsAppClient.GetChannelsInATeam(teamID, channelsQuery+")")
+			if err != nil {
+				p.API.LogDebug("Unable to get the MS Teams channel information for a team", "TeamID", teamID, "Error", err.Error())
```

Comment:
```suggestion
				p.API.LogDebug("Unable to get the MS Teams channel information for the team", "TeamID", teamID, "Error", err.Error())
```
`teamID` is specific here